### DISCUSSION TITLE
How to exclude internally used module and...requirements?

### DISCUSSION THREAD

**Discussion by SHxKM** [ansible-collections/overview#193] [Q&A] [answered]
We've just published `v1.0.0` thanks to a lot of help from this community, especially @tadeboro and @felixfontein.

One thing I don't like is the fact that there are two plugins listed for the collection:

![Screen Shot 2021-12-14 at 15 28 57](https://user-images.githubusercontent.com/6573474/146007412-e00a4ee3-af3a-42c2-996e-42de34226b74.png)

How can I exclude these from showing up? `spot_ansible_module` is supposed to be used **internally** by the modules and never directly by consumers, and re: `requirements` - I don't even know how it got there.

How can I exclude these from showing up?

**Comment by tadeboro** [comment, marked as answer]
It is not possible to tell Ansible Galaxy what files it should list. Right now, it lists just about everything under the `plugins` folder.

As for marking thins as private, see https://github.com/ansible-collections/overview/blob/main/collection_requirements.rst#standards-for-developing-module-and-plugin-utilities

**Reply by SHxKM** [reply]
> It is not possible to tell Ansible Galaxy what files it should list. Right now, it lists just about everything under the plugins folder.

Got it, so I couldn't have prevented these two from appearing. `doc_fragments` has to be in `plugins`, right? so I couldn't have prevented this one..?

> As for marking thins as private, see https://github.com/ansible-collections/overview/blob/main/collection_requirements.rst#standards-for-developing-module-and-plugin-utilities

Yep, real bummer here:

> It is a breaking change when you make an existing module_utils private and in that case the collection requires a major version bump.

Thanks!

**Reply by tadeboro** [reply]
> Got it, so I couldn't have prevented these two from appearing. `doc_fragments` has to be in `plugins`, right? so I couldn't have prevented this one..?

You did everything by the book and there is nothing more you can actually do to hide those files from Ansible Galaxy. So my advice would be "just ignore that file listing and go on with your life" ;)
